piksi_multi_cpp/src/receiver/receiver.cc

```diff
@@ -53,6 +53,10 @@ std::shared_ptr<Receiver> Receiver::create(
 }
 
 Receiver::~Receiver() {
+  // Close thread.
+  is_running_.store(false);
+  process_thread_.join();
```

Comment:
I would check if the thread is joinable (http://www.cplusplus.com/reference/thread/thread/joinable/), in case it already stopped due to an error or so.